docs/source/building.md

```diff
@@ -49,6 +49,20 @@ There are numerous device backends, options, and architecture-specific optimizat
 ````
 which activates the OpenMP backend. All the options controlling device backends, options, architectures, and third-party libraries (TPLs) are given in [CMake Keywords](../keywords).
 
+## Separate Compilation via CMake Language
+
+Kokkos supports separating the compilation of source files using Kokkos from others. This is controlled similar to a CMake language. The feature requires Kokkos to be compiled with the keyword `Kokkos_ENABLE_COMPILE_AS_CMAKE_LANGUAGE=ON`. The availability of the feature can be checked via `find_package(Kokkos COMPONENTS separable_compilation)`.
```

Comment:
Using the launch compiler is a lot more convenient when you don't care that it's used for all files anyway.

Reply:
We offer the `separable_compilation` component and it actually makes sense, that libs that require this feature can check if it is available in the used kokkos (which might be installed by an admin outside of user control).
The `find_package` checks if the component is enabled in the found kokkos ... thus I would rather call it a check vs a request. Does this make sense?

Due to the problems with `kokkos_compilation` the feature can only be used by manually setting the CUDA properties on the targets and the source files as @masterleinad pointed out correctly. Making a feature that has no performance portable way of use in a lib that is all about performance portability is not a good idea IMHO.

Reply:
> We offer the separable_compilation component and it actually makes sense, that libs that require this feature can check if it is available in the used kokkos (which might be installed by an admin outside of user control).
The find_package checks if the component is enabled in the found kokkos; thus, I would call it a check instead of a request. Do you think this makes sense?

I think I understand your point, but I think CMake can be understood the other way around. For example, I would likely write:
```CMake
find_package(Kokkos COMPONENTS separable_compilation REQUIRED)
```
Here, I expect `find_package` to do the Kokkos look-up and select the `separable_compilation` component. (and thus, it is why I think `request` is a good fit to describe that).
I do not mean to check for Kokkos that have already been found, and I do not think CMake works this way. When using boost, for example, COMPONENTS are additive.

> Making a feature with no performance portable way of use in a lib that is all about performance portability is not a good idea IMHO.

We are speaking about CMake, and I am not sure there is any link with *performance* portability (but I agree with the portability part).

However, nowadays, people will expect a library that provides a CMake interface to behave the CMake way, i.e., letting CMake decide the compiler depending on the source properties, for Cuda, HIP, ...
I agree it is out of scope for this PR, but I naively thought that the `kokkos_compilation` function would be portable even if it is a no-op when using the wrapper.

Reply:
merged the suggestion